### ISSUE TITLE
Actual , Expected values for a failed expect/assert

### ISSUE THREAD

**Issue by lazy-dude** [Snaipe/Criterion#389]
In case of failure of some expects/asserts I want to see the actual and expected values that differ. I know of the printf like capability of expects/asserts. But automated informing the coder of these values is a better option IMHO.

**Comment by Snaipe** [comment]
Is this fixed by the new assert API on the `bleeding` branch? `cr_assert(eq(u32, 1, 2))` typically diffs the actual vs expected values.

**Comment by lazy-dude** [comment]
Consider this example:

``` C
#include <stdlib.h>


int rangeBitwiseAnd(int left, int right)
{
    int r=~0;
    int i;
    
    for(i=left; i<=right; i++)
    {
        r = r&i;
        if(r==0)
            return 0;
    }
    return r;
}
```

``` C
#include <criterion/criterion.h>
#include <criterion/new/assert.h>
#include "try.h"

TestSuite(suite1);

Test(suite1, test1)
{
    int output=rangeBitwiseAnd(5,7);
    cr_expect_eq(output,4, "o is %d\n",output);
}

Test(suite1, test2)
{
    cr_expect_eq(rangeBitwiseAnd(0,0),0);
}

Test(suite1, test3)
{
    cr_expect_eq(rangeBitwiseAnd(1,2147483647),0);
}
```

**Comment by Snaipe** [comment]
cr_expect_eq is part of the old assert API -- `cr_expect(eq(int, rangeBitwiseAnd(left, right), 0)` would be the one that I'd expect to diff on error.

**Comment by lazy-dude** [comment]
Would you please tell where is documentation for new API ?

**Comment by Snaipe** [comment]
You need to go to the [latest](https://criterion.readthedocs.io/en/latest/assert.html) docs rather than the stable/master docs, since the former represents what's currently on the bleeding branch

**Comment by Snaipe** [comment]
I think this is fixed by 2.4.0
